testsuite/classlibrary/TestSoundFile.sc

```diff
@@ -48,9 +48,93 @@ TestSoundFile : UnitTest {
 		// collect all instance variables into a Dictionary, remove problematic 'fileptr'
 		instVars = soundFile.getSlots.asDict;
 		instVars.removeAt(\fileptr);
-		
+
 		this.assertEquals(instVars, info, "SoundFile information should match that of actual sound file");
 
 	}
 
+	// tests several features:
+	// openRead and openWrite class methods (not tested above)
+	// readData and writeData
+	test_readWrite {
+		var path = PathName.tmp +/+ "SoundFile-test%.aiff".format(this.hash.asHexString(8)),
+		file, written = Signal.fill(100, { 1.0.rand2 }), read;
+
+		file = SoundFile.openWrite(path, "aiff", "float", 1, 44100);
+		if(file.isNil) {
+			this.assert(false, "SoundFile.openWrite into temp directory should not fail");
+		};
+		try {
+			file.writeData(written);
+		} { |error|
+			this.assert(false, "SoundFile:writeData should not fail with error " ++ error.errorString);
+		};
+		file.close;
+
+		file = SoundFile.openRead(path);
+		this.assertEquals(file.numFrames, written.size, "SoundFile size should match data size written by writeData");
+
+		try {
+			read = Signal.newClear(file.numFrames);
+			file.readData(read);
+		} { |error|
+			this.assert(false, "SoundFile:readData should not fail with error " ++ error.errorString);
+		};
+		file.close;
+
+		this.assertEquals(written, read, "SoundFile:readData should get the same data that were written");
+
+		File.delete(path);
```

Comment:
this should be split into several tests - it looks like 4, one each for openRead, openWrite, readData, writeData. Checking the file descriptor is OK but doesn't strictly need to happen here, it would be easy to trace that back if it is a problem.

Reply:
Sorry but I disagree. You can't test readData without writing data, and you can't test writing data if file opening fails.

The idea of "one and only one test per method" is, IMO, incompatible with complex tests that have chains of actions where success at one stage depends on success in all previous stages.

Alternately, if these are to be split up, then we need a mechanism to ensure that the order is always openWrite, writeData, openRead, readData, and that you don't tear-down the tests at each step. Does UnitTest provide such a mechanism?

Reply:
Or rather, instead of stating an opinion, let me ask what is the recommended code structure for complex sequences in tests.

Reply:
It's fine, we can leave this.

Reply:
Ok, appreciate the understanding. I did actually think about splitting them up, but decided that it would result either in a confusing code structure, or code duplication.

Agreed that it isn't ideal to pack too many tests into one method, but sometimes it's the least-bad option.

I'll address the other changes later today, or maybe tomorrow.